### PR TITLE
[WNMGDS-530] Add icon to error messages

### PR DESCRIPTION
**Pull request by bernardwang**
### Summary
- Adds an error icon next to form error messages via CSS 

### How to test
- View form component changes on doc site, run through with screenreaders (it should be the same)
- Run the HC DS in a consuming application like App3 or Windowshop to ensure no obvious conflicts.

### CODE REVIEW

File: src/styles/components/_FormLabel.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+.ds-c-field__error-message::before {
+	background-image: url('#{$image-path}/error-red.svg');
+  background-repeat: no-repeat;
+  background-size: $error-message-icon-size;
+  background-position-y: center;
+  background-position-x: left;
+	padding-right: $error-message-icon-size;
+	margin-left: -5px;
+	content: ''
+}
```
Comment (nichia) [inline src/styles/components/_FormLabel.scss:12]: Add a new line at end of file.

**Comment by line47** [comment]
I'm also thinking about using purely CSS by using pseudo elements `::before` and `::after` on the `.ds-c-field__error-message` to add the circle and the exclamation point. 🤔 

This would be easier for modifying the color and sizing of both the circle and the exclamation point itself... 
I did a bit of exploring this and something along these lines could work for the styles

```
.ds-c-field__error-message {
  position: relative;
  padding-left:24px;
}
.ds-c-field__error-message::before {
  content: '';
  background-color: red;
  height: 18px;
  width: 18px;
  display: inline-block;
  border-radius: 100%;
  vertical-align: baseline;
  margin-right: 5px;
  position: absolute;
  left: 0;
  top: 3px;
}

.ds-c-field__error-message::after {
  content: '!';
    color: white;
    left: 7px;
    position: absolute;
    font-weight: bold;
}
```
and it looks good in Chrome and IE11 with some quick testing 
Screen Shot 2020-12-04 at 1 18 03 PM
Screen Shot 2020-12-04 at 1 18 14 PM

**Comment by bernardwang** [comment]
> 1. I noticed on the date field html examples the error wraps and looks odd, I'm not sure why it is wrapping.

I'm not seeing [this on my end](https://gyazo.com/fd8018a512591107dfffc326a4bab174), what browser did you use to get that? This PR doesnt change any of the form spacing or styling, so whatever we see should be the existing styles.

> 2. Related - should we handle the wrapping of the error message differently? Now that there is an icon I'm wondering if the icon should always be out to the left and have the text always aligned on the right

I don't feel super strongly on this, I think it makes sense as an inline icon or as a separately aligned element. Considering it is styled using pseudo elements, its easier to have it inline.

> I'm also thinking about using purely CSS by using pseudo elements `::before` and `::after` on the `.ds-c-field__error-message` to add the circle and the exclamation point. 🤔

What's the advantage of using CSS to draw the icon? I chose to use `background-image` because its simpler and less code and because its 100% aligned with the provided svg icon.

**Comment by line47** [comment]
1. I'm seeing the error message wrap on the date field page in Chrome and Safari (this is only happening with the HTML examples and is not happening with the React examples

2. Yeah let's leave it as it and see if folks have issues with it

3. I'm fine with leaving as a background image - was thinking it would be easier for folks to change if it was CSS as recreating custom images can be tricky.

**Comment by line47** [comment]
Looks like adding an icon for the inverse examples is needed and then it should be good to go!

**Comment by bernardwang** [comment]
@line47 @nichia This should be ready for a final review!

**Review by nichia — APPROVED** [review]
Great work!

**Pull request merged**